src/popup/router/pages/FungibleTokens/TokenDetails.vue

```diff
@@ -52,44 +53,136 @@
       class="token-info"
     >
       <DetailsRow
-        :label="$t('pages.token-details.abbreviation')"
+        :label="$t('pages.token-details.token')"
         :text="tokenData.symbol"
       />
       <DetailsRow
+        v-if="tokenData.decimals"
         :label="$t('pages.token-details.decimals')"
         :text="tokenData.decimals"
       />
       <DetailsRow
         v-if="tokenData.contract"
-        :class="{ contract: tokenData.contract }"
         :label="$t('pages.token-details.contract')"
-        :text="tokenData.contract"
-      />
+      >
+        <template #text>
+          <AddressShortening :address="tokenData.contract" />
+        </template>
+      </DetailsRow>
       <DetailsRow
-        :label="$t('pages.token-details.available-supply')"
+        v-if="tokenData.circulating_supply"
+        :label="$t('pages.token-details.max-supply')"
         :text="tokenData.circulating_supply"
       />
       <DetailsRow
+        v-if="tokenData.total_supply"
         :label="$t('pages.token-details.total-supply')"
         :text="tokenData.total_supply"
       />
       <DetailsRow
-        :label="$t('pages.token-details.max-supply')"
-        :text="tokenData.max_supply"
+        v-if="tokenData.market_cap"
+        :label="$t('pages.token-details.market-cap')"
+        class="price"
+      >
+        <template #text>
+          {{ formatCurrency(tokenData.market_cap) }}
+        </template>
+      </DetailsRow>
+      <DetailsRow
+        v-if="!tokenData.isAe"
+        label="Holders"
       />
-      <DetailsRow :label="$t('pages.token-details.price-ae')" />
       <DetailsRow
-        :label="$t('pages.token-details.price')"
-        :text="tokenData.current_price ? formatCurrency(tokenData.current_price) : ''"
+        v-if="tokenPairs.length > 1"
+      >
```

Comment:
It can be in one line
```suggestion
      <DetailsRow v-if="tokenPairs.length > 1">
```